src/components/flat-table/flat-table-row/flat-table-row.component.tsx

```diff
@@ -102,47 +96,94 @@ export const FlatTableRow = React.forwardRef<
     }: FlatTableRowProps,
     ref
   ) => {
-    const internalId = useRef(id ?? guid());
+    const internalId = useRef(String(id ?? guid()));
     const [isExpanded, setIsExpanded] = useState(expanded);
     let rowRef = useRef<HTMLTableRowElement>(null);
     if (ref) {
       rowRef = ref as React.MutableRefObject<HTMLTableRowElement | null>;
     }
     const firstColumnExpandable = expandableArea === "firstColumn";
-    const [leftStickyCellWidths, setLeftStickyCellWidths] = useState<number[]>(
-      []
+    const [leftPositions, setLeftPositions] = useState<Record<string, number>>(
+      {}
     );
-    const [rightStickyCellWidths, setRightStickyCellWidths] = useState<
-      number[]
-    >([]);
-    const [leftPositions, setLeftPositions] = useState<number[]>([]);
-    const [rightPositions, setRightPositions] = useState<number[]>([]);
-    const childrenArray = useMemo(() => React.Children.toArray(children), [
+    const [rightPositions, setRightPositions] = useState<
+      Record<string, number>
+    >({});
+    const [firstCellIndex, setFirstCellIndex] = useState(0);
+    const [lhsRowHeaderIndex, setLhsRowHeaderIndex] = useState(-1);
+    const [rhsRowHeaderIndex, setRhsRowHeaderIndex] = useState(-1);
+    const [firstCellId, setFirstCellId] = useState<string | null>(null);
+    const [cellsArray, setCellsArray] = useState<Element[]>([]);
+    const [tabIndex, setTabIndex] = useState(-1);
+    let interactiveRowProps = {};
+
+    useLayoutEffect(() => {
+      const buildPositionMap = (cells: Element[]) =>
+        cells.reduce((acc: Record<string, number>, cell, index) => {
+          const currentId = cells[index].getAttribute("id");
+          if (currentId) {
+            if (index === 0) {
+              acc[currentId] = 0;
+            } else {
+              const previousId = cells[index - 1].getAttribute("id");
+              if (previousId) {
+                acc[currentId] =
+                  acc[previousId] +
+                  (cells[index - 1] as HTMLTableCellElement).offsetWidth;
+              }
+            }
+          }
+          return acc;
+        }, {});
+
+      const cells = rowRef.current?.querySelectorAll("th, td");
+
+      if (cells) {
+        const cellArray = Array.from(cells);
+        setCellsArray(cellArray);
+
+        const firstIndex = cellArray.findIndex(
+          (cell) =>
+            cell.getAttribute("data-component") !== "flat-table-checkbox"
+        );
+        const lhsIndex = cellArray.findIndex(
+          (cell) => cell.getAttribute("data-sticky-align") === "left"
+        );
+        const rhsIndex = cellArray.findIndex(
+          (cell) => cell.getAttribute("data-sticky-align") === "right"
+        );
+
+        if (firstIndex !== -1) {
```

Comment:
**comment** (non-blocking): similar to the previous comment, I feel we should be setting these things to `undefined` or some similar "empty" value if the corresponding index is -1.

Reply:
Fair point, I'll see if returning them to the initial (default) values works here as I'd prefer not to update the types given there's a lot going on already in this component, if not I'll set them to undefined in the else branches